automation/check-patch.k8s-1.14.2-kind-sriov.sh

```diff
@@ -0,0 +1,252 @@
+set -x
+
+#############################################################
+# CI Notes TODO Move in a more appropriate location
+# Each CI node labeled as sriov-enable should have the following setup in place:
+# - have the vfio_pci module loaded with a modprobe.d rule
+# - since we use kind, in order to have the dns working we need to
+#   - have the br_netfilter module loaded with a modprobe.d rule
+#   - have the following lines on /etc/sysctl.conf (see https://stackoverflow.com/questions/48148838/kube-dns-error-reply-from-unexpected-source)
+#               net.bridge.bridge-nf-call-ip6tables = 1
+#               net.bridge.bridge-nf-call-iptables = 1
+#               net.bridge.bridge-nf-call-arptables = 1
+#
+#############################################################
+
+
+#############################################################
+# This is based on https://github.com/SchSeba/kubevirt-docker
+#############################################################
+
+export NO_PROXY="localhost,127.0.0.1,172.17.0.2"
+
+export WORKSPACE="${WORKSPACE:-$PWD}"
+readonly ARTIFACTS_PATH="${ARTIFACTS-$WORKSPACE/exported-artifacts}"
+
+CLUSTER_NAME=sriov-ci
+CLUSTER_CONTROL_PLANE=${CLUSTER_NAME}-control-plane
+CONTAINER_REGISTRY_HOST="localhost:5000"
+
+CLUSTER_CMD="docker exec -it -d ${CLUSTER_CONTROL_PLANE}"
+
+KUBEVIRT_PATH=`pwd`
+CLUSTER_DIR="cluster-up/cluster/k8s-1.14.2-kind-sriov"
+MANIFESTS_DIR="${CLUSTER_DIR}/manifests"
+
+
+function wait_containers_ready {
+    # wait until all containers are ready
+    while [ -n "$(kubectl get pods --all-namespaces -o'custom-columns=status:status.containerStatuses[*].ready,metadata:metadata.name' --no-headers | grep false)" ]; do
+        echo "Waiting for all containers to become ready ..."
+        kubectl get pods --all-namespaces -o'custom-columns=status:status.containerStatuses[*].ready,metadata:metadata.name' --no-headers
+        sleep 10
+    done
+}
+
+# NOTE: this assumes that once at least a single virt- service pops up then
+# others will pop up too in quick succession, at least before the first one
+# transits to ready state. If it's ever not the case, we may end up exiting
+# this function before all virt pods are scheduled and in ready state. If this
+# ever happens, we may need to list all services we expect in a kubevirt
+# cluster and check each of them is up and running.
+function wait_kubevirt_up {
+    # it takes a while for virt-operator to schedule virt pods; wait for at least one of them to pop up
+    while [ -z "$(kubectl get pods -n kubevirt | grep virt)" ]; do
+        echo "Waiting for all pods to create ..."
+        kubectl get pods -n kubevirt | grep virt
+	sleep 10
+    done
+
+    wait_containers_ready
+}
+
+function wait_kind_up {
+    while [ -z "$(docker exec --privileged ${CLUSTER_CONTROL_PLANE} kubectl --kubeconfig=/etc/kubernetes/admin.conf get nodes --selector=node-role.kubernetes.io/master -o=jsonpath='{.items..status.conditions[-1:].status}' | grep True)" ]; do
```

Comment:
Maybe missing space as print$1

Reply:
Not sure I understood where

Reply:
There is no awk anymore so all good.

Reply:
I suggest waiting until the DNS nodes are ready, that will mean that networking is ready too

`kubectl wait -n kube-system --for=condition=Ready -l k8s-app=kube-dns pods`

the control plane can be up but the CNI plugin and other components can take longer and add flakiness

Reply:
actually, IIRC I took that command from `kind` debug output. That's what v0.3.0 calling to check if the node is ready (https://github.com/kubernetes-sigs/kind/blob/194cbdec9ab1bae59037e4ee98d3d90688f97d12/pkg/cluster/nodes/nodes.go#L106) . 
I could not wait to be ready because dns wasn't starting without flannel, so I had to split the waiting part out, install flannel and then wait, but that change sounds reasonable.

Reply:
yep, at least in my experience, when DNS pods are ready the cluster is fully functional to start to test